packages/devtools_app/lib/src/screens/memory/memory_allocation_table_view.dart

```diff
@@ -80,6 +80,15 @@ class AllocationTableViewState extends State<AllocationTableView>
     super.didChangeDependencies();
     if (!initController()) return;
 
+    // Make sure that we actually initialize the tracker tree rather than
+    // relying on `controller.updateClassStackTraces` to be updated.
+    // TODO: find out why AllocationTableView isn't persisting when switching
```

Comment:
Ah that fixes it, although I just noticed we're doing something weird with the state management for the table displaying the actual tracking information since it doesn't update when you add new classes from the list. I tried messing around with it a bit, and it seems like we're not calling `TreeTable._initData()` in `didUpdateWidget()` since the `collectionEquals(widget.dataRoots, oldWidget.dataRoots) == true)`. Not sure why `oldWidget.dataRoots.length` is the same as `widget.dataRoots.length` if we're rebuilding. We're going to want to look at that in a separate PR.